src/distribution/weibull.rs

```diff
@@ -308,14 +309,10 @@ impl Continuous<f64, f64> for Weibull {
     fn ln_pdf(&self, x: f64) -> f64 {
         if x < 0.0 {
             f64::NEG_INFINITY
-        } else if x == 0.0 && self.shape == 1.0 {
```

Comment:
can we add these two paths back in, the `0.0 - ` is probably unnecessary but the way they were written is to prevent unnecessary work or divisions

Reply:
I've also removed other specialisations, do you want me to put those back in too?
I'll have to profile it. It gets rid of divisions but more logarithms have to be evaluated. An additional division could be left out by writing `x.ln() - self.scale.ln()`.

Reply:
I don't see any other ones that were removed outside the ones that are no longer relevant due to removals of `f64::INFINITY` but good point about `ln`. Do you mind running a quick benchmark? Should be pretty simple using https://github.com/bluss/bencher/, here's a toy example https://github.com/boxtown/counter/blob/master/benches/bench.rs from another repo

Reply:
In the formulas of `pdf` and `ln_pdf` I removed the specialisation `shape = 1`.

My preference is towards predictability. Trying to manage special cases leads to uglier code with hard to predict control flow. As such tricks accumulate it gets increasingly hard to understand what's going on, with the interactions with caches and branch prediction. On some machines it may become quicker, on some slower.

Reply:
```
running 2 tests
test weibull_new ... bench:         110 ns/iter (+/- 1)
test weibull_old ... bench:         116 ns/iter (+/- 1)
```

Reply:
```
running 4 tests
test weibull_new_x_equals_0 ... bench:           4 ns/iter (+/- 0)
test weibull_new_x_equals_2 ... bench:         110 ns/iter (+/- 4)
test weibull_old_x_equals_0 ... bench:          25 ns/iter (+/- 0)
test weibull_old_x_equals_2 ... bench:         117 ns/iter (+/- 8)
```
Here's the bench if you want to run it.
```
#[macro_use]
extern crate bencher;

extern crate statrs;

use bencher::Bencher;
use std::f64::*;
use statrs::distribution::{Weibull,Continuous};

struct WeibullNew(Weibull);

impl WeibullNew {
    fn ln_pdf(&self, x: f64) -> f64 {
        if x < 0.0 {
            NEG_INFINITY
        } else if x == INFINITY {
            NEG_INFINITY
        } else {
            self.0.pdf(x).ln()
        }
    }
}

fn weibull_old_x_equals_0(bench: &mut Bencher) {
    let w = Weibull::new(1.0, 0.5).unwrap();
    bench.iter(|| {
        w.ln_pdf(0.0);
    })

}

fn weibull_new_x_equals_0(bench: &mut Bencher) {
    let w = WeibullNew(Weibull::new(1.0, 0.5).unwrap());
    bench.iter(|| {
        w.ln_pdf(0.0);
    })
}

fn weibull_old_x_equals_2(bench: &mut Bencher) {
    let w = Weibull::new(1.0, 0.5).unwrap();
    bench.iter(|| {
        w.ln_pdf(2.0);
    })

}

fn weibull_new_x_equals_2(bench: &mut Bencher) {
    let w = WeibullNew(Weibull::new(1.0, 0.5).unwrap());
    bench.iter(|| {
        w.ln_pdf(2.0);
    })
}

benchmark_group!(first, weibull_old_x_equals_0, weibull_new_x_equals_0);
benchmark_group!(second, weibull_old_x_equals_2, weibull_new_x_equals_2);
benchmark_main!(first,second);
```